packages/runtime-core/src/hydration.ts

```diff
@@ -672,6 +678,7 @@ export function createHydrationFunctions(
           vnode.type,
         )
       logMismatchError()
+      handleMismatchHook(vnode, parentComponent, node)
```

Comment:
If `handleMismatchHook` is dev-only, then it should be:
```ts
__DEV__ && handleMismatchHook(vnode, parentComponent, node)
```
it's more treeshaking friendly.